src/components/calcite-panel/calcite-panel.tsx

```diff
@@ -180,45 +362,130 @@ export class CalcitePanel {
       />
     ) : null;
 
-    const slotNode = <slot name={SLOTS.headerTrailingContent} />;
+    const slotNode = <slot name={SLOTS.headerActionsEnd} />;
+    const hasEndActions = getSlotted(el, SLOTS.headerActionsEnd);
 
-    return (
-      <div class={CSS.headerTrailingContent} key="header-trailing-content">
+    return hasEndActions || dismissibleNode ? (
+      <div
+        class={{ [CSS.headerActionsEnd]: true, [CSS.headerActions]: true }}
+        key="header-actions-end"
+      >
         {slotNode}
         {dismissibleNode}
       </div>
+    ) : null;
+  }
+
+  renderMenuItems(): VNode {
+    const { el, menuOpen, menuButtonEl } = this;
+
+    return (
+      <calcite-popover
+        disablePointer={true}
+        flipPlacements={["bottom-end", "top-end"]}
+        offsetDistance={0}
+        onKeyDown={this.menuActionsKeydown}
+        open={menuOpen}
+        placement="bottom-end"
+        referenceElement={menuButtonEl}
+        theme={getElementTheme(el)}
```

Comment:
I'm not sure we need to pass the theme anymore here. Now that we're sharing theme styling? Is that correct? We can probably remove a lot of places where we say `theme={...`

Reply:
I can remove it.